src/recipes.jl

```diff
@@ -331,6 +336,16 @@ function Makie.plot!(gp::GraphPlot)
     return gp
 end
 
+function prep_arguments(o::Observable, size::Observable, default::Observable)
+    @lift begin
+        if isscalar($o)
+            isnothing($o) ? $default : $o
+        else
+            [getattr(o, i, $default) for i in 1:$size]
+        end
+    end
+end
+
```

Comment:
Should `prep_arguments` maybe forward arguments given as a plain vector rather than looping through them and creating a new one?

pros: prob faster
cons: not possible to denote default by using `nothing` inside vector (not sure if thats a feature^^); If we loop through it it is possible to pass a "longer" vector than needed. this might help with adjusting the graph size which is allways annoying.

Reply:
> cons: not possible to denote default by using nothing inside vector 

when passing in a typeof `AbstractVector` it looks like there is no codepath to use any default value. Or is it ? Can you please elaborate ?

> this might help with adjusting the graph size

Yeah. I don't know how often people actually do it, but then it will be at least theoretically possible.

Reply:
> this might help with adjusting the graph size

To take the best of both worlds probably probably we need to write more complicated logic. something like
```julia
    @lift begin
        if isscalar($o)
            return isnothing($o) ? $default : $o
        elseif $o isa AbstractVector
            length($indices) == length($o) && return $o
        end
        return [getattr(o, i, $default) for i in $indices]
    end
```

But currently I have the impression that will still break if the size of graph is changed, since in the `AbstractVector` case in `getattr` no expansion is done with any default value.
So basically this logic needs to enter in the `getattr`:
```julia
function getattr(x, idx, default=nothing)
    if x isa AbstractVector && !isa(x, Point)
        return idx < length(x) ? x[idx] : default
    elseif x isa DefaultDict || x isa DefaultOrderedDict
...
```

Reply:
On this pr, if i pass `v=[1, nothing, 3]` it will call `[getattr(e, i, default) for e in v]` which will replace `nothing` by the default of the recipe inside `getattr`. But I think this isnt really a feature. I think i'd vote for adding
```julia
elseif $o isa Vector
    $o
else
```

For adjusting the graph size using a default dict might be the best option anyway, i like that.

Reply:
Which introduces a check in each iteration plus the allocation of new vector due to the comprehension. I am wondering whether we can avoid both by mutating the `AbastractVector` with `push!` ( or `deleteteat!` if graph decreases in size) but this will probably need even more complicated logic. 
Or something like `vcat(o$, fill(default, residualsize))` (or `deleteat!(o$, length(vertices(graph)):length($o)` for decreasing graph size).

Reply:
But I am wondering are there use cases for changing the graph size ? I mean doing `add_vertex!` or `add_edge!` don't even trigger the Observable calls. One should substitute the graph as a whole.

Reply:
We have #92 which is about changing graph size. However switching out the graph is something more complicated and need attantion throughout the entire recipe and would be a job for another PR so lets not overcomplicate things here. I think i prefer the explicitness of passing a vector directly without meddling to much.
```
elsif $o isa AbstractVector
    o[]
else
```
will explicitly fail in both cases: `length(o) > nv(g)` and `lenght(o) < nv(g)`. I like that over the current state of the PR where it works if the passed vector is to long because we only loop through and basicially select a subset of the given vector.